samplePDF/samplePDFFDBase.h

```diff
@@ -238,18 +231,40 @@ class samplePDFFDBase :  public samplePDFBase
   //===========================================================================
   //DB Vectors to store which kinematic cuts we apply
   //like in XsecNorms but for events in sample. Read in from sample yaml file 
-  std::vector< std::string > SelectionStr; 
-  std::vector< std::vector<double> > Selection; //The enum, then the bounds corresponding to the string
-
-  // like in XsecNorms but for events in sample. Read in from sample yaml file
-  // in samplePDFExperimentBase.cpp
-  std::vector< std::vector<double> > SelectionBounds;
-
   //What gets used in IsEventSelected, which gets set equal to user input plus 
   //all the vectors in StoreSelection
-  //std::vector< std::vector<double> > Selection;
+  /// @brief the Number of selections in the 
   int NSelections;
-  //What gets pulled from config options
-  std::vector< std::vector<double> > StoredSelection; 
-  //===========================================================================
+  
+  /// @brief What gets pulled from config options, these are constant after loading in
+  /// this is of length 3: 0th index is the value, 1st is lower bound, 2nd is upper bound
+  std::vector< std::vector<double> > StoredSelection;
+  /// @brief the strings grabbed from the sample config specifying the selections
+  std::vector< std::string > SelectionStr; 
+  /// @brief the bounds for each selection lower and upper
+  std::vector< std::vector<double> > SelectionBounds;
+  /// @brief a way to store selection cuts which you may push back in the get1DVar functions
+  /// most of the time this is just the same as StoredSelection
+  std::vector< std::vector<double> > Selection;
+   //===========================================================================
+
+  manager* SampleManager;
+  void InitialiseSingleFDMCObject(int iSample, int nEvents);
+  void InitialiseSplineObject();
+
+  double Unity = 1.;
+  double Zero = 0.;
+  
+  std::vector<std::string> mtuple_files;
+  std::vector<std::string> spline_files;
+  std::vector<int> sample_vecno;
+  std::vector<int> sample_oscnutype;
+  std::vector<int> sample_nutype;
+  std::vector<bool> sample_signal;
+
+  std::string mtupleprefix;
+  std::string mtuplesuffix;
+  std::string splineprefix;
+  std::string splinesuffix;
+
```

Comment:
Are these needed?
https://github.com/mach3-software/MaCh3/blob/5fa87ded750b3ba2f040e7eddd9e51e7dcc7350b/samplePDF/samplePDFFDBase.cpp#L135
Here you use them locally.

So stuff defined in header is really useless

Reply:
The spline prefixes are actually used here; https://github.com/mach3-software/MaCh3/blob/5fa87ded750b3ba2f040e7eddd9e51e7dcc7350b/samplePDF/samplePDFFDBase.cpp#L1606